### PR TITLE
Add MSBuild 17.4.0 SBRPs

### PR DESCRIPTION
**Pull request by mmitche**
Required for aspnetcore prebuilt detection.

### CODE REVIEW

File: eng/Build.props [hunk 1]
```diff
@@ -167,6 +167,13 @@
     <DependencyPackageProjects Include="$(RepoRoot)src\referencePackages\src\**\System.Configuration.Configurationmanager.7.0.0.csproj" />
     <DependencyPackageProjects Include="$(RepoRoot)src\referencePackages\src\**\System.Security.Cryptography.Xml.7.0.1.csproj" />
     <DependencyPackageProjects Include="$(RepoRoot)src\referencePackages\src\**\System.Threading.Channels.6.0.0.csproj" />
+
+    <DependencyPackageProjects Include="$(RepoRoot)src\referencePackages\src\**\Microsoft.Win32.Registry.5.0.0.csproj" />
+    <DependencyPackageProjects Include="$(RepoRoot)src\referencePackages\src\**\Microsoft.Build.Framework.17.4.0.csproj" />
+    <DependencyPackageProjects Include="$(RepoRoot)src\referencePackages\src\**\Microsoft.NET.StringTools.17.4.0.csproj" />
+    <DependencyPackageProjects Include="$(RepoRoot)src\referencePackages\src\**\Microsoft.Build.Utilities.Core.17.4.0.csproj" />
+    <DependencyPackageProjects Include="$(RepoRoot)src\referencePackages\src\**\System.Collections.Immutable.6.0.0.csproj" />
```
Comment (MichaelSimons) [inline eng/Build.props:175]: This is an existing package.

File: eng/Build.props [hunk 2]
```diff
@@ -167,6 +167,13 @@
     <DependencyPackageProjects Include="$(RepoRoot)src\referencePackages\src\**\System.Configuration.Configurationmanager.7.0.0.csproj" />
     <DependencyPackageProjects Include="$(RepoRoot)src\referencePackages\src\**\System.Security.Cryptography.Xml.7.0.1.csproj" />
     <DependencyPackageProjects Include="$(RepoRoot)src\referencePackages\src\**\System.Threading.Channels.6.0.0.csproj" />
+
+    <DependencyPackageProjects Include="$(RepoRoot)src\referencePackages\src\**\Microsoft.Win32.Registry.5.0.0.csproj" />
```
Comment (MichaelSimons) [inline eng/Build.props:171]: This is an existing package.  Entry is not necessary here.

Comment (mmitche) [reply eng/Build.props:171]: I see...are the packages not in this list built first?

Comment (MichaelSimons) [reply eng/Build.props]: Only new package that aren't in previously source-built packages are in this list.  After preview 4 is release and we update the artifacts this list will get cleared out.

**Review by MichaelSimons — APPROVED** [review]

**Pull request merged**